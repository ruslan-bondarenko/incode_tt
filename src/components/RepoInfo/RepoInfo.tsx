import { IRepoInfo, capitalizeWords } from "@/shared";
import { Breadcrumb } from "antd";
import React, { FC } from "react";

type Props = {
  info: IRepoInfo;
};

const RepoInfo: FC<Props> = ({ info }) => {
  return (
    <div style={{ padding: "0 2rem" }}>
      <Breadcrumb
        items={[
          {
            title: (
              <a
                href={`https://github.com/${info.name}`}
                target="_blank"
                rel="noreferrer"
              >
                {capitalizeWords(info.name)}
              </a>
            ),
          },
          {
            title: (
              <a href={info.repo.url} target="_blank" rel="noreferrer">
                {capitalizeWords(info.repo.name)}
              </a>
            ),
          },
        ]}
      />
    </div>
  );
};

export default RepoInfo;
