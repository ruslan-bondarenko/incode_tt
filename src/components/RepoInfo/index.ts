export {default as RepoInfo} from './RepoInfo';