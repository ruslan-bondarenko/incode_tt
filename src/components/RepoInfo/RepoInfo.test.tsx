import { render, screen } from "@testing-library/react";
import React from "react";

import RepoInfo from "./RepoInfo";
import { describe, it } from "@jest/globals";

declare const expect: jest.Expect;

const mockedInfo = {
  name: "owner",
  repo: {
    name: "repo",
    url: "https://github.com/owner/repo",
  },
};
describe("RepoInfo component", () => {
  it("RepoInfo renders", () => {
    render(<RepoInfo info={mockedInfo} />);

    setTimeout(() => {
      expect(screen.getByTestId("repoinfo")).toBeInTheDocument();
    }, 0);
  });

  it("RepoInfo snapshot", () => {
    const view = render(<RepoInfo info={mockedInfo} />);

    expect(view).toMatchSnapshot();
  });
});
