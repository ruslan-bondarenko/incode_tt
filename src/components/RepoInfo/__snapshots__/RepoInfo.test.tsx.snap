// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoInfo component RepoInfo snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="repoinfo"
        style="padding: 0px 2rem;"
      >
        <nav
          class="ant-breadcrumb css-dev-only-do-not-override-1kuana8"
        >
          <ol>
            <li>
              <span
                class="ant-breadcrumb-link"
              >
                <a
                  href="https://github.com/owner"
                  rel="noreferrer"
                  target="_blank"
                >
                  Owner
                </a>
              </span>
            </li>
            <li
              aria-hidden="true"
              class="ant-breadcrumb-separator"
            >
              /
            </li>
            <li>
              <span
                class="ant-breadcrumb-link"
              >
                <a
                  href="https://github.com/owner/repo"
                  rel="noreferrer"
                  target="_blank"
                >
                  Repo
                </a>
              </span>
            </li>
          </ol>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="repoinfo"
      style="padding: 0px 2rem;"
    >
      <nav
        class="ant-breadcrumb css-dev-only-do-not-override-1kuana8"
      >
        <ol>
          <li>
            <span
              class="ant-breadcrumb-link"
            >
              <a
                href="https://github.com/owner"
                rel="noreferrer"
                target="_blank"
              >
                Owner
              </a>
            </span>
          </li>
          <li
            aria-hidden="true"
            class="ant-breadcrumb-separator"
          >
            /
          </li>
          <li>
            <span
              class="ant-breadcrumb-link"
            >
              <a
                href="https://github.com/owner/repo"
                rel="noreferrer"
                target="_blank"
              >
                Repo
              </a>
            </span>
          </li>
        </ol>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
